src/Kiota.Builder/SearchProviders/Apicurio/ApicurioSearchProvider.cs

```diff
@@ -0,0 +1,146 @@
+    {
+        ArgumentNullException.ThrowIfNull(httpClient);
+        ArgumentNullException.ThrowIfNull(configuration);
+        ArgumentNullException.ThrowIfNull(logger);
+        _httpClient = httpClient;
+        _logger = logger;
+        _configuration = configuration;
+        _authenticatedAuthenticationProvider = authenticatedAuthenticationProvider;
+        KeysToExclude = new(StringComparer.OrdinalIgnoreCase);
+    }
+
+    public async Task<IDictionary<string, SearchResult>> SearchAsync(string term, string? version, CancellationToken cancellationToken)
+    {
+        if (_configuration.ApiBaseUrl == null)
+        {
+            _logger.LogInformation("Apicurio provider not configured, skipping.");
+            return EMPTY_RESULT;
+        }
+
+        var authenticationProvider = _authenticatedAuthenticationProvider != null ?
+            _authenticatedAuthenticationProvider :
+            new AnonymousAuthenticationProvider();
+        using var requestAdapter = new HttpClientRequestAdapter(authenticationProvider, httpClient: _httpClient);
+        requestAdapter.BaseUrl = _configuration.ApiBaseUrl.AbsoluteUri;
+        var apicurioClient = new ApicurioClient(requestAdapter);
+
+        IDictionary<string, SearchResult> result;
+        try
+        {
+            ArtifactSearchResults? searchResults = await apicurioClient.Search.Artifacts.GetAsync(config =>
+            {
+                config.QueryParameters.Limit = _configuration.ArtifactsLimit;
+                config.QueryParameters.Offset = 0;
+                switch (_configuration.SearchBy)
+                {
+                    case ApicurioConfiguration.ApicurioSearchBy.LABEL:
+                        config.QueryParameters.Labels = new string[] { term };
+                        break;
+                    case ApicurioConfiguration.ApicurioSearchBy.PROPERTY:
+                        config.QueryParameters.Properties = new string[] { term };
+                        break;
+                }
+            }, cancellationToken).ConfigureAwait(false);
+
+            if (searchResults == null || searchResults!.Artifacts == null)
+                return EMPTY_RESULT;
+
+            var dictionaries = searchResults!.Artifacts!.Select(async x =>
+            {
+                var groupId = (x.GroupId != null) ? x.GroupId : "default";
+                var uiUrl = new Uri(_configuration.UIBaseUrl + "/artifacts/" + groupId + "/" + x.Id);
+                var restUrl = new Uri(_configuration.ApiBaseUrl + "/groups/" + groupId + "/artifacts/" + x.Id);
+
+                Console.WriteLine("FOUND SOMETHING: " + groupId + " - version: " + version);
+
+                if (!string.IsNullOrEmpty(version))
+                {
+                    var versionMetadata = await apicurioClient.Groups[groupId].Artifacts[x.Id].Versions[version!].Meta.GetAsync().ConfigureAwait(false);
```

Comment:
the cancellation token should be passed here

Reply:
done 👍